### ISSUE TITLE
Nextcloud 11 login test not working

### ISSUE THREAD

**Issue by poVoq** [codinguser/gnucash-android#651]
#### Steps to reproduce the behaviour
1. Go to setting and back up on owncloud
2. Input nextcloud11 server adress(http) and credentials
3. When pressing test it says oc server ok, but login details incorrect

#### Expected behaviour
Nextcloud 11 should still be fully compatible with owncloud, or did they change the login Api?

#### Software specifications
* GnuCash Android version: 2.1.3
* System Android version: 5.1
* Device type: x86 Asus Zenfone2

**Comment by sehucke** [comment]
Same with version 2.1.5 on a Moto G 4G (peregrine). Neither user/password nor user/app-pin currently works.

**Comment by codinguser** [comment]
@poVoq can you use the new Save As.. functionality in v2.2.0 with your NextCloud successfully?

**Comment by wenxin-wang** [comment]
Seems that GnuCash is using `/index.php/ocs/cloud/user` for logging-in, but at least on NextCloud 11 the `GET` request will get 302 redirected to the login page (`/login`)

**Comment by codinguser** [comment]
@wenxin-wang are you referring to the "Save As.." functionality, or the logging in via ownCloud integration?

**Comment by sehucke** [comment]
You can use the "Send to..." export target for use with the Nextcloud app. It is not that much convenient (you have to choose the upload folder each time) but it works.

IMHO, this is a more general solution and should work with other cloud storages as well - as long as they provide an Android client. ;-)

**Comment by Johnjustdoit** [comment]
Did you set a new APP-Pin for gnucash in the nextcloud settings in the webpanel?

**Comment by Johnjustdoit** [comment]
Go to the nextcloud webpanel->press your nickname in the top right-> personal.
Then set a new APP-PIN and copy the new passwort. Paste the passwort into the gnucash panel with your nextcloud Nickname and your Serveradress.

**Comment by poVoq** [comment]
AFAIK the app pin should only be necessary if an external auth system like SAML or LDAP is used in Nextcloud and thus no password is stored in the Nextcloud database.

**Comment by sehucke** [comment]
@poVoq @Johnjustdoit : An app pin is necessary if you activated TOTP for your user. AFAIK, it is fully transparent to the client if you enter your password or an app pin.

Just do as follows:
1. Log into your Nextcloud Android client with your password.
1. Activate TOTP via web interface.
1. Watch your Android client mourn about invalid credentials.
1. Create an app pin and use that pin instead of your password.
1. Watch your Android client successfully connect to your NC instance.

That means it won't make any difference for GnuCash.

BTW: Has anyone tried to use GnuCash app with NC12?

**Comment by Johnjustdoit** [comment]
Thank you for clearing it up!
I found a workaround to sync it automatically in my private network. (Should also work with a nextcloud reachable in the www)
I use the app Foldersync Pro (3,19€) and the app tasker (2,99€). First I set a periodic export in GnuCash to my SD Card. Then I set a task in tasker to sync all my previous added syncronisations(GnuCash, Photos,...etc) in FolderSync when my phone log into my home WLAN. This is posible for any folder and file on your phone. On my PC I use the nextcloudapp to use the file in my nextcloud folder with GnuCash.

**Comment by sehucke** [comment]
Yep, that should do the trick. :-)

Nevertheless, the NC Android client v1.5 (and above) has the potential to render this toolchain obsolete. See nextcloud/android#285 (it is a meta ticket gathering the work done in separate tickets).

**Comment by denysvitali** [comment]
I'm having the exact same problem with Nextcloud 11.0.3. @sehucke tip didn't work (I don't have 2FA enabled atm, which may be the cause)

**Comment by sehucke** [comment]
Which tip did you mean @denysvitali ? I suppose you are referring to https://github.com/codinguser/gnucash-android/issues/651#issuecomment-315597218 ? My explanation of app pin usage was not meant to be a tip for getting GnuCash to work with NC. I tried to explain the exact opposite: _It does not matter if you use your password or an app pin inside GnuCash. The client does **not** know there is a difference._

At the moment there seem to be only two ways to get GnuCash working with NC:
1. To use "Send to..." as described in https://github.com/codinguser/gnucash-android/issues/651#issuecomment-312599156 or...
1. ... to set up a toolchain like @Johnjustdoit explained in https://github.com/codinguser/gnucash-android/issues/651#issuecomment-315598825. (Which I have not tested so far by myself. Should work anyway.)

Besides that: App pins should work regardless of whether you have enabled 2FA or not. If I remember correctly, I have successfully tried that before enabling 2FA in my NC installation.
You can easily verify if your app pins work correctly. Just generate a new pin and use an app to log in with it (NC, DAVdroid, CalDAV, CardDAV...). However, atm this won't help you with this GnuCash problem.

**Comment by denysvitali** [comment]
I'm already working on a fix to integrate NextCloud into GnuCash...
I thought NC and OC had the same authentication mechanism

**Comment by 5nafu** [comment]
With Nextcloud 12 I could successfully work around the Issue with an entry in my webserver config (in this case apache2 with virtualhosts):
```
RewriteEngine On
RewriteRule ^/index.php/ocs/cloud/user /ocs/v1.php/cloud/user
```

**Comment by rivaldi8** [comment]
Fixed develop branch (fc261f7).
